test/functional/test-preconnect.js

```diff
@@ -39,133 +60,207 @@ describe('preconnect', () => {
-    expect(document.querySelectorAll('link[rel=preconnect]'))
-        .to.have.length(0);
-    preconnect.url('https://y.preconnect.com/foo/bar');
-    expect(document.querySelectorAll('link[rel=preconnect]'))
-        .to.have.length(0);
-    clock.tick(180 * 1000);
-    preconnect.url('https://y.preconnect.com/foo/bar');
-    expect(document.querySelectorAll('link[rel=preconnect]'))
-        .to.have.length(1);
+    return getPreconnectIframe().then(iframe => {
+      preconnect.url('https://y.preconnect.com/foo/bar',
+          /* opt_alsoConnecting */ true);
+      expect(iframe.doc.querySelectorAll('link[rel=preconnect]'))
+          .to.have.length(1);
+      clock.tick(10000);
+      expect(iframe.doc.querySelectorAll('link[rel=preconnect]'))
+          .to.have.length(0);
+      preconnect.url('https://y.preconnect.com/foo/bar');
+      expect(iframe.doc.querySelectorAll('link[rel=preconnect]'))
+          .to.have.length(0);
+      clock.tick(180 * 1000);
+      preconnect.url('https://y.preconnect.com/foo/bar');
+      expect(iframe.doc.querySelectorAll('link[rel=preconnect]'))
+          .to.have.length(1);
+    });
   });
 
   it('should prefetch', () => {
-    preconnect.prefetch('https://a.prefetch.com/foo/bar');
-    preconnect.prefetch('https://a.prefetch.com/foo/bar');
-    preconnect.prefetch('https://a.prefetch.com/other');
-    preconnect.prefetch(javascriptUrlPrefix + ':alert()');
-    // Also preconnects.
-    expect(document.querySelectorAll('link[rel=dns-prefetch]'))
-        .to.have.length(1);
-    expect(document.querySelector('link[rel=dns-prefetch]').href)
-        .to.equal('https://a.prefetch.com/');
-    expect(document.querySelectorAll('link[rel=preconnect]'))
-        .to.have.length(1);
-    expect(document.querySelector('link[rel=preconnect]').href)
-        .to.equal('https://a.prefetch.com/');
-    // Actual prefetch
-    const fetches = document.querySelectorAll(
-        'link[rel=prefetch]');
-    expect(fetches).to.have.length(2);
-    expect(fetches[0].href).to.equal('https://a.prefetch.com/foo/bar');
-    expect(fetches[1].href).to.equal('https://a.prefetch.com/other');
+    return getPreconnectIframe().then(iframe => {
+      preconnect.prefetch('https://a.prefetch.com/foo/bar');
+      preconnect.prefetch('https://a.prefetch.com/foo/bar');
+      preconnect.prefetch('https://a.prefetch.com/other');
+      preconnect.prefetch(javascriptUrlPrefix + ':alert()');
+      // Also preconnects.
+      expect(iframe.doc.querySelectorAll('link[rel=dns-prefetch]'))
+          .to.have.length(1);
+      expect(iframe.doc.querySelector('link[rel=dns-prefetch]').href)
+          .to.equal('https://a.prefetch.com/');
+      expect(iframe.doc.querySelectorAll('link[rel=preconnect]'))
+          .to.have.length(1);
+      expect(iframe.doc.querySelector('link[rel=preconnect]').href)
+          .to.equal('https://a.prefetch.com/');
+      // Actual prefetch
+      const fetches = iframe.doc.querySelectorAll(
```

Comment:
I had that in mind and I was thinking of always mocking the `perloadSupported=false` except when I am testing for `preloading`. What do you think? So basically this will change the `preloadSupported=undefined` to false in the `beforeEach` block?